pkg/stack/unwind/unwind_table.go

```diff
@@ -199,29 +199,38 @@ func (ptb *UnwindTableBuilder) PrintTable(writer io.Writer, file *objectfile.Obj
 	return nil
 }
 
-func ReadFDEs(file *objectfile.ObjectFile) (frame.FrameDescriptionEntries, elf.Machine, error) {
+func ReadFDEs(file *objectfile.ObjectFile, ehFrame bool) (frame.FrameDescriptionEntries, elf.Machine, error) {
 	obj, err := file.ELF()
 	if err != nil {
 		return nil, elf.EM_NONE, fmt.Errorf("failed to open elf: %w", err)
 	}
-	defer obj.Close()
 
 	arch := obj.Machine
 
-	sec := obj.Section(".eh_frame")
+	var secName string
+	if ehFrame {
+		secName = ".eh_frame"
+	} else {
+		secName = ".debug_frame"
+	}
+	sec := obj.Section(secName)
 	if sec == nil {
 		return nil, arch, ErrEhFrameSectionNotFound
 	}
 
 	// TODO: Consider using the debug_frame section as a fallback.
 	// TODO: Needs to support DWARF64 as well.
-	ehFrame, err := sec.Data()
+	secData, err := sec.Data()
 	if err != nil {
-		return nil, arch, fmt.Errorf("failed to read .eh_frame section: %w", err)
+		return nil, arch, fmt.Errorf("failed to read %s section: %w", secName, err)
 	}
 
 	// TODO: Byte order of a DWARF section can be different.
-	fdes, err := frame.Parse(ehFrame, obj.ByteOrder, 0, pointerSize(obj.Machine), sec.Addr, arch)
+	var ehFrameAddr uint64
+	if ehFrame {
+		ehFrameAddr = sec.Addr
```

Comment:
Interesting!   My musl changes don't do this, I wonder if .debug_frame's in .debug files are different, what clued you in that you needed to do this?

Reply:
I included this change in my musl stuff and the eh-frame output didn't change, probably good to include but apparently the .debug_frame's in the musl libc doesn't include anything different that the eh/debug conditionals trigger on.